app/src/main/java/com/android/unio/ui/settings/Settings.kt

```diff
@@ -52,35 +62,76 @@ fun SettingsScreen(navigationAction: NavigationAction) {
       }
 }
 
+@OptIn(ExperimentalPermissionsApi::class)
 @Composable
 fun SettingsContainer() {
   val context = LocalContext.current
+  val preferences by LocalPreferenceFlow.current.collectAsState()
+
+  /** Location Permissions * */
+  val locationPermissions =
+      rememberMultiplePermissionsState(
+          permissions =
+              listOf(
+                  Manifest.permission.ACCESS_FINE_LOCATION,
+                  Manifest.permission.ACCESS_COARSE_LOCATION))
+  val requestPermissionLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.RequestMultiplePermissions()) {
+          permissions ->
+        locationPermissions.launchMultiplePermissionRequest()
+      }
+
+  /** Language * */
+  val language = preferences.get<String>(AppPreferences.LANGUAGE) ?: AppPreferences.Language.default
+  val locale = Locale(language)
+  Locale.setDefault(locale)
+
+  val configuration = context.resources.configuration
+  configuration.setLocale(Locale(language))
+  configuration.setLayoutDirection(locale)
+
+  context.createConfigurationContext(configuration)
+  context.resources.updateConfiguration(configuration, context.resources.displayMetrics)
 
   ProvidePreferenceLocals(flow = LocalPreferenceFlow.current) {
     LazyColumn(
         modifier = Modifier.testTag("SettingsContainer"),
     ) {
       listPreference(
-          modifier = Modifier.testTag(PreferenceKeys.THEME),
-          key = PreferenceKeys.THEME,
+          modifier = Modifier.testTag(AppPreferences.THEME),
+          key = AppPreferences.THEME,
           title = { Text(context.getString(R.string.settings_theme_title)) },
-          valueToText = { AnnotatedString(it) },
-          summary = { Text(it) },
-          values = listOf(Theme.LIGHT, Theme.DARK, Theme.SYSTEM),
-          defaultValue = Theme.SYSTEM,
+          summary = { Text(AppPreferences.Theme.toDisplayText(it)) },
+          valueToText = { AnnotatedString(AppPreferences.Theme.toDisplayText(it)) },
+          values = AppPreferences.Theme.asList,
+          defaultValue = AppPreferences.Theme.default,
           icon = {
             Icon(
                 imageVector =
                     when (it) {
-                      Theme.DARK -> Icons.Default.Nightlight
-                      Theme.LIGHT -> Icons.Default.WbSunny
+                      AppPreferences.Theme.DARK -> Icons.Default.Nightlight
+                      AppPreferences.Theme.LIGHT -> Icons.Default.WbSunny
                       else -> Icons.Default.Smartphone
                     },
                 contentDescription = context.getString(R.string.settings_theme_content_description))
           })
+      listPreference(
+          modifier = Modifier.testTag(AppPreferences.LANGUAGE),
```

Comment:
Should we really use the class's object as a test tag? Perhaps separating test tags from object strings might be a good convention but as it avoids creating more string objects I don't think it's that big of a deal.